core/src/ast_builder/index_by.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{
+    ast::IndexItem,
+    ast_builder::{SelectNode, TableFactorNode},
+};
+
+#[derive(Clone, Debug)]
+pub struct IndexNode<'a> {
```

Comment:
it would be enough with `pub enum IndexItem` which contains `ast::IndexItem` and some nodes in `index_item/**` because `TableFactorNode` can have `IndexItemNode`.